File: SPA/FrontEnd/src/app/Models/hospedaje/hospedaje.ts
import { PuntoTuristico } from '../Punto/punto';
import { Imagen } from '../Imagen/imagen';
import { Resenia } from '../Resenia/resenia';
export class Hospedaje {
    id: number;
    nombreHospedaje: string;
    punto: PuntoTuristico;
    cantidadEstrellas: number;
    direccion: string;
    imagenes: Imagen[];
    precioPorNoche: number;
    precioTotalPeriodo: number;
    descripcion: string;
    ocupado: boolean;
    capacidad: number;
    resenias: Resenia[];
    puntaje: number;

    constructor(id: number, punto: PuntoTuristico, nombre: string, direccion: string,
        cantidadEstrellas: number, ocupado: boolean, resenias: Resenia[],
        imagenes: Imagen[], precioPorNoche: number, precioTotalPeriodo: number, descripcion: string,
        capacidad: number, puntaje: number)
        {
            this.precioTotalPeriodo = precioTotalPeriodo;
            this.descripcion = descripcion;
            this.capacidad = capacidad;
            this.puntaje = puntaje;
            this.imagenes = imagenes;
            this.cantidadEstrellas = cantidadEstrellas;
            this.punto = punto;
            this.direccion = direccion;
            this.precioPorNoche = precioPorNoche;
            this.resenias = resenias;
            this.nombreHospedaje = nombre;
            this.ocupado = ocupado;
            this.id = id;
        }
}

File: Domain/Hospedaje.cs
using System;
using BuisnessLogic.Domain;

using System.Collections.Generic;

namespace Domain
{
    public class Hospedaje
    {
        public int Id { get; set; }
        public string NombreHospedaje { get; set; }
        public virtual PuntoTuristico PuntoTuristico { get; set; }
        public int CantidadEstrellas { get; set; }
        public string Direccion { get; set; }
        public virtual List<Imagen> Imagenes { get; set; }
        public int PrecioPorNoche { get; set; }
        public int PrecioTotalPeriodo { get; set; }
        public string Descripcion { get; set; }
        public bool Ocupado { get; set; }
        public int Capacidad { get; set; }
        public virtual List<Resenia> Resenias { get; set; }
        public double Puntaje { get; set; }
    }
}
